internal/pkg/manifest/http.go

```diff
@@ -38,6 +38,11 @@ func (r *RoutingRuleConfigOrBool) UnmarshalYAML(value *yaml.Node) error {
 	if !r.RoutingRuleConfiguration.IsEmpty() {
 		// Unmarshalled successfully to r.RoutingRuleConfiguration, unset r.Enabled, and return.
 		r.Enabled = nil
+		// this assignment lets us treat the main listener rule and additional listener rules equally
+		// because we eliminate the need for TargetContainerCamelCase by assigning its value to TargetContainer.
+		if r.TargetContainerCamelCase != nil {
```

Comment:
noice, we tested this right with something like `svc package` to make sure it works as expected?

If so maybe we can add a comment to the body of the PR saying that it resolves #4589

Reply:
Actually I tested it separately if it assigns the value of `TargetContainerCamelCase` to `targetContainer` and that worked well. But now when I tried running with svc package then we are running into error that cannot define both `target_container` and `targetContainer` which we can solve by setting the value of `TargetContainerCamelCase` to nil but it makes me think if we should modify the customer input or not? What do you think about it?

Reply:
oh go for it! It's okay to change it to `nil`